app/models/user.rb

```diff
@@ -27,6 +27,34 @@ def scoped_service_providers
       sort_by! { |sp| sp.friendly_name.downcase }
   end
 
+  def user_deletion_history
+    PaperTrail::Version
+      .where(event: 'destroy', item_type: 'UserTeam')
+      .where("object ->>'user_id' = '?'", id)
+  end
+
+  def user_deletion_report_item(record)
+    {
+      user_id: record[0]['user_id'],
+      user_email: User.find_by(id: record[0]['user_id'])&.email,
+      team_id: record[0]['group_id'],
+      team_name: Team.find_by(id: record[0]['group_id'])&.name,
+      removed_at: record[1],
+      whodunnit_id: record[2],
+      whodunnit_email: User.find_by(id: record[2])&.email
+    }
+  end
+
+  def user_deletion_history_report
+    user_deletion_history
+      .order(created_at: :desc)
+      .limit(5000)
+      .pluck(:object, :created_at, :whodunnit)
+      .map { |record|
+        user_deletion_report_item(record)
+      }
```

Comment:
I hope rubocop will get mad at this, but if not, I will.... I prefer braces syntax for blocks on the same line and `do...end` for multiline blocks so this should be one of two things:


```suggestion
      .map { |record| user_deletion_report_item(record) }
```

```suggestion
      .map do |record|
        user_deletion_report_item(record)
      end
```